lib/revert.js

```diff
@@ -23,222 +19,108 @@ class Revert {
-                    if (!revertToFeature) return callback(new Error('Feature to revert to is undefined'));
-
-                    if (!this.hasFeatureChanged(revertToFeature, currentFeature)) return cb();
-                    if (revertAllFeatures || _.isEqual(currentFeature, feature)) {
-                        this.revertFeature(currentFeature, feature, revertToFeature, (err, rolBackFeature) => {
-                            if (err) return callback(err);
-                            modifiedFeatureCollection.push(rolBackFeature);
-                            cb();
-                        });
-                    } else {
-                        // Ask the action for each feature that has changed
-                        this.askForAction(response.body, (err, action) => {
-                            if (err) return callback(err);
-                            if (action === 'ka') return callback(null, modifiedFeatureCollection);
-                            else if (action === 'r' || action === 'ra') {
-                                if (action === 'ra') revertAllFeatures = true;
-                                this.revertFeature(currentFeature, feature, revertToFeature, (err, rolBackFeature) => {
-                                    if (err) return callback(err);
-                                    modifiedFeatureCollection.push(rolBackFeature);
-                                });
-                            }
-                            cb();
-                        });
-                    }
-                });
-            });
-        }, (err) => {
-            if (err) return callback(err);
-            callback(null, modifiedFeatureCollection);
-        });
-    }
 
-    getRevertToFeature(historyList, feature, callback) {
-        if (!historyList) return callback(new Error('The historyList cannot be empty.'));
-        if (!feature) return callback(new Error('The feature cannot be empty.'));
+        async function main() {
+            if (!options.start) return cb(new Error('start delta is required'));
+            if (!options.end) return cb(new Error('start delta is required'));
+            if (options.start > options.end) return cb(new Error('start delta must be less than end delta'));
+            if (!options.output) return cb(new Error('output stream is required'));
 
-        let revertToFeature;
+            try {
+                const db = await revert.cache(options, self.api);
 
-        for (let i = 0; i < historyList.length; i++) {
-            if (_.isEqual(historyList[i].feat, feature)) {
-                if (feature.action === 'create') {
-                    revertToFeature = JSON.parse(JSON.stringify(feature));
-                    revertToFeature.geometry = null;
-                    revertToFeature.properties = {};
-                } else revertToFeature = historyList[i + 1].feat;
-                break;
+                revert.iterate(db, options.output);
```

Comment:
Should this clean up the database file before exiting?

Reply:
Good call, added a cleanup method